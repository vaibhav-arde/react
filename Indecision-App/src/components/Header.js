import React from 'react';

const Header = (props) => (
    <div>
        <h1>{props.title}</h1>
        {props.subTitle && <h2>{props.subTitle}</h2>}
    </div>
);
Header.defaultProps = {
    title: 'Indecision App!!'
}
// class Header extends React.Component {
//     render() {
//         // console.log(this.props)
//         return (
//             <div>
//                 <h1>{this.props.title}</h1>
//                 <h2>{this.props.subTitle}</h2>
//             </div>
//         );
//     }
// }

export default Header;