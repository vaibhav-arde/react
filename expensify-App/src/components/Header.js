import React from 'react';
import { Link, NavLink } from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <h3><Link to='/' > Dashboard</Link></h3>
        <h3><Link to='/create' > Create Expense</Link></h3>
        <h3><Link to='/edit' > Edit Expense</Link></h3>
        <h3><Link to='/help' > Get Help</Link></h3>

        <NavLink to='/' activeClassName='is-active' exact={true}> Dashboard</NavLink>
        <NavLink to='/create' activeClassName='is-active' > Create Expense</NavLink>
        <NavLink to='/edit' activeClassName='is-active' > Edit Expense</NavLink>
        <NavLink to='/help' activeClassName='is-active' > Get Help</NavLink>
    </header>
)

export default Header;