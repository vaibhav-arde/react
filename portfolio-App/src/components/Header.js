import React from 'react';
import { Link, NavLink } from 'react-router-dom'

const Header = () => (
    <header>
        <h1>PORTFOLIO</h1>

        <NavLink to='/' activeClassName='is-active' exact={true}> Home Page</NavLink>
        <NavLink to='/portfolio' activeClassName='is-active' > Portfolio </NavLink>
        <NavLink to='/contactPage' activeClassName='is-active' > Contact Page </NavLink>
    </header>
)

export default Header;