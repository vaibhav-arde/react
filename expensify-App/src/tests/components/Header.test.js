import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from '../../components/Header';

// below example test is using react-test-renderer
test('Should render Header correctly using react-test-renderer', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Header />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
})

// Below test are using enzyme by airbnb
test('Should render Header correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('h1').length).toBe(1);
    expect(wrapper.find('h1').text()).toBe('Expensify');
    // expect(wrapper).toMatchSnapshot();
    expect(toJSON(wrapper)).toMatchSnapshot();
});