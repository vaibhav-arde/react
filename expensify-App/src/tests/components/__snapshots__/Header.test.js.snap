// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <h3>
    <Link
      to="/"
    >
       Dashboard
    </Link>
  </h3>
  <h3>
    <Link
      to="/create"
    >
       Create Expense
    </Link>
  </h3>
  <h3>
    <Link
      to="/edit/:id"
    >
       Edit Expense
    </Link>
  </h3>
  <h3>
    <Link
      to="/help"
    >
       Get Help
    </Link>
  </h3>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
     Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
     Create Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/edit/:id"
  >
     Edit Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/help"
  >
     Get Help
  </NavLink>
</header>
`;

exports[`Should render Header correctly using react-test-renderer 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <h3>
    <Link
      to="/"
    >
       Dashboard
    </Link>
  </h3>
  <h3>
    <Link
      to="/create"
    >
       Create Expense
    </Link>
  </h3>
  <h3>
    <Link
      to="/edit/:id"
    >
       Edit Expense
    </Link>
  </h3>
  <h3>
    <Link
      to="/help"
    >
       Get Help
    </Link>
  </h3>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
     Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
     Create Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/edit/:id"
  >
     Edit Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/help"
  >
     Get Help
  </NavLink>
</header>
`;
